### PR TITLE
Add support for putting option types on the D-Bus as tuples.

### PR DESCRIPTION
**Pull request by mulkieran**
Use the support for the quota part of filesystem's spec.

Signed-off-by: mulhern <amulhern@redhat.com>

### CODE REVIEW

**Comment by mulkieran** [comment]
@agrover @trgill review?

This is a pretty straightforward change. I concluded that a tuple was a much better choice than a variant type in order to handle these kinds of optional values. There are related PRs in stratisd-client-dbus and stratis-cli.

File: src/sim_engine/filesystem.rs [hunk 1]
```diff
@@ -15,12 +17,12 @@ pub struct SimFilesystem {
...
 }
 impl SimFilesystem {
-    pub fn new_filesystem(name: &str, mount_point: &str, size: u64) -> SimFilesystem {
+    pub fn new_filesystem(name: &str, mount_point: &str, size: Option<u64>) -> SimFilesystem {
         SimFilesystem {
             name: name.to_owned(),
             uuid: Uuid::new_v4(),
             mount_point: mount_point.to_owned(),
-            size: size,
+            size: size.unwrap_or(DEFAULT_FILESYSTEM_SIZE),
```
Comment (trgill) [inline src/sim_engine/filesystem.rs]: If this unwrap fails, shouldn't we return an error?

Comment (trgill) [inline src/sim_engine/filesystem.rs]: Talked to @mulkieran in person - if this unwrap returns None, we want to set the default.  Looks good.

**Comment by trgill** [comment]
:+1:

**Comment by agrover** [comment]
Since `size` is really `quota_size`?, could you rename it to avoid any confusion, either in this PR, or open an issue to address this in a later PR?

I also have questions about mount_point but I've opened #155 to discuss independently.

Finally, quotas are slated for 2.0 and are going to be some work, so I've opened #154 to discuss that, too 😄

**Comment by mulkieran** [comment]
@agrover review?

**Comment by agrover** [comment]
👍

**Pull request merged**